src/de/hhn/aib/labsw/blackmirror/controller/API/websockets/MirrorApiWebsockets.java

```diff
@@ -1,42 +1,44 @@
-package de.hhn.aib.labsw.blackmirror.controller.API;
+package de.hhn.aib.labsw.blackmirror.controller.API.websockets;
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import de.hhn.aib.labsw.blackmirror.model.ApiDataModels.Location;
+import de.hhn.aib.labsw.blackmirror.controller.API.MirrorApi;
+import de.hhn.aib.labsw.blackmirror.controller.API.TopicListener;
 import org.java_websocket.WebSocket;
 import org.java_websocket.handshake.ClientHandshake;
 import org.java_websocket.server.WebSocketServer;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Iterator;
+import java.util.*;
 
 /**
  * MirrorApiServer provides an easy to use Interface to the App.
+ * Author: Luis Gutzeit
+ * Version: 1.0
  */
-public class MirrorApiServer extends WebSocketServer {
+public class MirrorApiWebsockets extends WebSocketServer implements MirrorApi {
     //https://mvnrepository.com/artifact/org.java-websocket/Java-WebSocket/1.5.3
-
-    ArrayList<WebSocket> sessions = new ArrayList<>();
+    List<WebSocket> sessions = new ArrayList<>();
     ObjectMapper mapper = new ObjectMapper();
-    HashMap<String, ArrayList<TopicListener>> listeners = new HashMap<>();
-    private static MirrorApiServer instance = null;
+    Map<String, List<TopicListener>> listeners = new HashMap<>();
+
+    private static MirrorApiWebsockets instance = null;
 
     /**
      * private constructor (singleton pattern)
+     * singleton pattern is necessary because multiple instances could conflict with each other
```

Comment:
its not necessary to use a singleton here
the design pattern has pretty heavy flaws and is considered an anti-pattern by some (https://en.wikipedia.org/wiki/Singleton_pattern#Criticism), especially preventing unit testing in this particular case

why cant multiple instances exist with each other? If there are hardware limitations i'm sure the underlying system will throw an corresponding exception
also: just because a class is not a singleton doesnt mean you have to have more than one instance of it

for now we can leave it as a singleton until we find a way to write appropriate unit tests later where this class must be changed to no longer be a singleton